File: OpenVehicleApp/src/com/openvehicles/OVMS/ui/utils/MarkerGenerator.java
package com.openvehicles.OVMS.ui.utils;

import com.google.android.gms.maps.model.LatLng;

import com.androidmapsextensions.GoogleMap;
import com.androidmapsextensions.MarkerOptions;

public class MarkerGenerator {
	public static void addMarkers(GoogleMap map, String title, String snippet, LatLng latlng, String cpid) {
		MarkerOptions options = new MarkerOptions();
		map.addMarker(options.title(title).snippet(snippet).position(latlng).data(cpid));
	}
}

File: app/src/main/java/com/openvehicles/OVMS/ui/utils/MarkerGenerator.kt
package com.openvehicles.OVMS.ui.utils

import com.androidmapsextensions.GoogleMap
import com.androidmapsextensions.MarkerOptions
import com.google.android.gms.maps.model.LatLng

object MarkerGenerator {

    @JvmStatic
    fun addMarkers(
        map: GoogleMap,
        title: String?,
        snippet: String?,
        latLng: LatLng?,
        cpId: String?
    ) {
        map.addMarker(
            MarkerOptions()
                .title(title)
                .snippet(snippet)
                .position(latLng)
                .data(cpId)
        )
    }
}
